pkg/errors/factory.go

```diff
@@ -0,0 +1,34 @@
+package errors
+
+import (
+	api_errors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+)
+
+func isCRDError(status api_errors.APIStatus) bool {
+	for _, cause := range status.Status().Details.Causes {
+		if strings.HasPrefix(cause.Message, "404")  && cause.Type == v1.CauseTypeUnexpectedServerResponse {
+			return true
+		}
+	}
+
+	return false
+}
+
+func Build(err error) error {
```

Comment:
I created a error package to deal with this, it returns a valid error interface + the APIStatus object as well